### ISSUE TITLE
HTTP code 500 when unique constraint violated

### ISSUE THREAD

**Issue by kevinprotoss** [arangodb/arangodb#1458]
Hi,

I got an error from ArangoDB

```
{ [ArangoError: unique constraint violated (while executing)]
  name: 'ArangoError',
  message: 'unique constraint violated (while executing)',
  errorNum: 1210,
  code: 500,
  stack: 'ArangoError: unique constraint violated (while executing)\n    at new ArangoError (/Invend/int_modules/restore-server/node_modules/arangojs/lib/error.js:10:13)\n    at /Invend/int_modules/restore-server/node_modules/arangojs/lib/connection.js:91:71\n    at /Invend/int_modules/restore-server/node_modules/arangojs/lib/util/request.js:81:12\n    at /Invend/int_modules/restore-server/node_modules/arangojs/lib/util/once.js:7:15\n    at IncomingMessage.<anonymous> (/Invend/int_modules/restore-server/node_modules/arangojs/lib/util/request.js:89:11)\n    at IncomingMessage.emit (events.js:129:20)\n    at _stream_readable.js:908:16\n    at process._tickDomainCallback (node.js:381:11)'}
```

The error is understandable, but the code of ArangoError is a little bit strange. The `code` should be the http status code, shouldn't it? If a request for creating duplicate resource occurs, the `code` should be 409, Conflict instead, right?

Version: 2.6.3

Bg,
Kevin

**Comment by jsteemann** [comment]
You're right, it should be 409. What server-side action is your code calling?

**Comment by kevinprotoss** [comment]
I ensured an index like so

```
"indexes": [
  {
    "indexType": "hash",
    "fields": ["subject", "creator"],
      "unique": true
  }
]
```

Then made an `INSERT` to create a new document. There is already a document with the same subject and creator in the DB, therefore I expect an 409 error.

The log of queries are the following

```
Tue, 25 Aug 2015 13:11:27 GMT gss-provider:arango AQL INSERT @node INTO @@collection LET nodes = NEW RETURN nodes
Tue, 25 Aug 2015 13:11:27 GMT gss-provider:arango bindVars { '@collection': 'pin',
  node: 
   { '@id': 'http://www.xxx.com/pins/f32d9d457619462e8318efd14c4ab06b',
     '@type': 'http://www.xxx.com/classes/Pin#CreatePinOperation',
     subject: 'http://www.xxx.com/elements/luftschlange5invitation',
     created: '2015-08-25T13:11:27.392Z',
     creator: 'http://www.xxx.com/users/xxx',
     modified: '2015-08-25T13:11:27.392Z',
     _key: 'f32d9d457619462e8318efd14c4ab06b' } }
```

**Comment by jsteemann** [comment]
Fixed in 2.6 and devel.
